K2Bridge/Models/Request/Aggregations/Bucket/DateRange/DateRangeAggregation.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Models.Request.Aggregations
+{
+    using System.Collections.Generic;
+    using K2Bridge.JsonConverters;
+    using K2Bridge.Visitors;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Bucket aggregation that creates one bucket per value range of a field.
+    /// </summary>
+    internal class DateRangeAggregation : BucketAggregation
+    {
+        /// <summary>
+        /// Gets or sets the field to aggregate.
+        /// </summary>
+        [JsonProperty("field")]
+        public string Field { get; set; }
+
+        /// <summary>
+        /// Gets or sets the range expressions.
+        /// </summary>
+        [JsonProperty("ranges")]
+        public IList<DateRangeAggregationExpression> Ranges { get; set; }
+
+        /// <summary>
+        /// Gets or sets the keyed flag, indicating the bucket should be returned as a hash.
+        /// </summary>
+        [JsonProperty("keyed")]
+        public bool Keyed { get; set; }
+
+        /// <summary>
+        /// Gets or sets the time zone.
+        /// </summary>
+        [JsonProperty("time_zone")]
+        public string TimeZone { get; set; }
```

Comment:
Is TimeZone really used ? If not, we can perhaps add a workitem to implement TimeZone behavior for both DateRange and DateHistogram.

Reply:
The time zone is ignored in general at the moment, it sounds like a good investigation user story to add to the backlog.